app/src/main/java/com/technopark/hw_1/NumberAdapter.java

```diff
@@ -0,0 +1,37 @@
+package com.technopark.hw_1;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+public class NumberAdapter extends RecyclerView.Adapter<NumberViewHolder> {
+    private final OnNumberClickListener mOnNumberClickListener;
+    private final DataSource mDataSource;
+
+    public NumberAdapter(DataSource dataSource, OnNumberClickListener onNumberClickListener) {
```

Comment:
Don't forget about `@NonNull` and `Nullable` annotations everywhere in java code. It makes life much simpler. Especially in case you are on big project with Java/Kotlin codebase - it allows you easily interop between two languages